Tests/FluentAssertions.Specs/Streams/StreamAssertionSpecs.cs

```diff
@@ -9,833 +9,853 @@ namespace FluentAssertions.Specs.Streams
+                act.Should().NotThrow();
+            }
 
-        #region HavePosition / NotHavePosition
-
-        public static IEnumerable<object[]> GetPositionExceptions()
-        {
-            // https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.position#exceptions
-            yield return new object[] { new IOException("GetPositionExceptionMessage") };
-            yield return new object[] { new NotSupportedException("GetPositionExceptionMessage") };
-            yield return new object[] { new ObjectDisposedException("GetPositionExceptionMessage") };
-        }
-
-        [Fact]
-        public void When_a_stream_has_the_expected_position_it_should_succeed()
-        {
-            // Arrange
-            using var stream = new TestStream { Seekable = true, Position = 10 };
+            [Fact]
+            public void When_having_a_readable_stream_be_not_readable_should_fail()
+            {
+                // Arrange
+                using var stream = new TestStream { Readable = true };
 
-            // Act
-            Action act = () =>
-                stream.Should().HavePosition(10);
+                // Act
+                Action act = () =>
+                    stream.Should().NotBeReadable("we want to test the failure {0}", "message");
 
-            // Assert
-            act.Should().NotThrow();
-        }
+                // Assert
+                act.Should().Throw<XunitException>()
+                    .WithMessage("Expected stream not to be readable *failure message*, but it was.");
+            }
 
-        [Fact]
-        public void When_a_stream_has_the_unexpected_position_it_should_fail()
-        {
-            // Arrange
-            using var stream = new TestStream { Seekable = true, Position = 1 };
+            [Fact]
+            public void When_null_not_be_readable_should_fail()
+            {
+                // Arrange
+                TestStream stream = null;
 
-            // Act
-            Action act = () =>
-                stream.Should().HavePosition(10, "we want to test the failure {0}", "message");
+                // Act
+                Action act = () =>
+                {
+                    using var _ = new AssertionScope();
+                    stream.Should().NotBeReadable("we want to test the failure {0}", "message");
+                };
 
-            // Assert
-            act.Should().Throw<XunitException>()
-                .WithMessage("Expected the position of stream to be 10* *failure message*, but it was 1*.");
+                // Assert
+                act.Should().Throw<XunitException>()
+                    .WithMessage("Expected stream not to be readable *failure message*, but found a <null> reference.");
+            }
         }
 
-        [Theory]
-        [MemberData(nameof(GetPositionExceptions))]
-        public void When_a_throwing_stream_should_have_a_position_it_should_fail(Exception exception)
+        public class PositionException
```

Comment:
The reason is this error: `xUnit1015: MemberData must reference an existing member 'GetLengthExceptions' on type 'FluentAssertions.Specs.Streams.StreamAssertionSpecs.NotHaveLength'`.
And since I do not like duplicateing this two methods `GetPositionsException` and `GetLengthException` to both classes [Not]HavePosition and [Not]HaveLength i've decided to go that way..

Reply:
It seems you can use
```cs
[MemberData(nameof(StreamAssertionsSpecs.GetPositionExceptions), MemberType = typeof(StreamAssertionsSpecs))]
```

(I'm not on my dev machine, so can't check it myself right now)

Reply:
ok.. I will check this soon

thanks for the hint :)

Reply:
done

Reply:
I learned something new as well :+1: